Add execute_timeout to GQL Client in Python SDK

mirnawong1 opened the pull request:
this pr adds a SL rn entry documenting a public change to the python sdk. added execute_timeout to GQL client in python sdk.

[linear](https://linear.app/dbt-labs/issue/SL-3244/add-execute-timeout-to-gql-client-in-python-sdk#comment-657bfcec)

<!-- vercel-deployment-preview -->
---
🚀 Deployment available! Here are the direct links to the updated files:


- https://docs-getdbt-com-git-python-timeout-dbt-labs.vercel.app/docs/dbt-versions/release-notes

<!-- end-vercel-deployment-preview -->

mirnawong1 committed:
add sl entry

mirnawong1 commented on website/docs/docs/dbt-versions/release-notes.md:
```diff
@@ -18,6 +18,7 @@ Release notes are grouped by month for both multi-tenant and virtual private clo
 
 ## February 2025
 
+- **Enhancement**: Improved stability in the [Python SDK](/docs/dbt-cloud-apis/sl-python) by increasing the default `execute_timeout` in GraphQL from 10s to 30s to reduce test flakiness (inconsistent results), particularly for the "execute query sync" endpoint. 
```
```suggestion
- **Enhancement**: The [Python SDK](/docs/dbt-cloud-apis/sl-python) has improved stability by increasing the default `execute_timeout` in GraphQL from 10s to 30s to reduce test flakiness (inconsistent results). This is particularly useful for the "execute query sync" endpoint. 
```

mirnawong1 committed:
Update website/docs/docs/dbt-versions/release-notes.md

mirnawong1 commented on website/docs/docs/dbt-versions/release-notes.md:
```diff
@@ -18,6 +18,7 @@ Release notes are grouped by month for both multi-tenant and virtual private clo
 
 ## February 2025
 
+- **Enhancement**: The [Python SDK](/docs/dbt-cloud-apis/sl-python) has improved stability by increasing the default `execute_timeout` in GraphQL from 10s to 30s to reduce test flakiness (inconsistent results). This is particularly useful for the "execute query sync" endpoint. 
```
@serramatutu is the timeout configurable or is it set to 30 secs?

mirnawong1 replied:
```suggestion
- **Enhancement**: The [Python SDK](/docs/dbt-cloud-apis/sl-python) has improved stability by increasing the default `execute_timeout` in GraphQL from 10 seconds to 30 seconds to reduce test flakiness (inconsistent results). This is particularly useful for the "execute query sync" endpoint. 
```

mirnawong1 committed:
Update website/docs/docs/dbt-versions/release-notes.md

mirnawong1 commented on website/docs/docs/dbt-versions/release-notes.md:
```diff
@@ -18,6 +18,7 @@ Release notes are grouped by month for both multi-tenant and virtual private clo
 
 ## February 2025
 
+- **Enhancement**: The [Python SDK](/docs/dbt-cloud-apis/sl-python) has improved stability by increasing the default `execute_timeout` in GraphQL from 10 seconds to 30 seconds to reduce test flakiness (inconsistent results). This is particularly useful for the "execute query sync" endpoint. 
```
```suggestion
- **Enhancement**: The [Python SDK](/docs/dbt-cloud-apis/sl-python) has improved stability by increasing the default `execute_timeout` in GraphQL from 10 seconds to 30 seconds to reduce test flakiness (inconsistent test results). This is particularly useful for the "execute query sync" endpoint. 
```

mirnawong1 committed:
Update website/docs/docs/dbt-versions/release-notes.md

mirnawong1 commented on website/docs/docs/dbt-versions/release-notes.md:
```diff
@@ -18,6 +18,7 @@ Release notes are grouped by month for both multi-tenant and virtual private clo
 
 ## February 2025
 
+- **Enhancement**: The [Python SDK](/docs/dbt-cloud-apis/sl-python) has improved stability by increasing the default `execute_timeout` in GraphQL from 10 seconds to 30 seconds to reduce test flakiness (inconsistent test results). This is particularly useful for the "execute query sync" endpoint. 
```
```suggestion
- **Enhancement**: The [Python SDK](/docs/dbt-cloud-apis/sl-python) has improved stability by increasing the default `execute_timeout` in GraphQL from 10 seconds to 30 seconds to reduce test flakiness (inconsistent test results). This is useful for endpoints like the "execute query sync". 
```

mirnawong1 committed:
Update website/docs/docs/dbt-versions/release-notes.md

nataliefiann reviewed (approved):
Hiya @mirnawong1 

Thanks for creating this. No suggestions from me. I've approved this for you.

Kind Regards
Natalie

serramatutu reviewed (commented):
Hey! So the test flakiness was happening internally and it didn't actually affect customers. What we changed in the SDK that does affect customers is we added a new "timeout" parameter which can be used to specify timeouts if customers want it.

They can either pass in a number or a "TimeoutOptions" in this parameter to control connect, execute and close timeouts granularly.

mirnawong1 commented:
> Hey! So the test flakiness was happening internally and it didn't actually affect customers. What we changed in the SDK that does affect customers is we added a new "timeout" parameter which can be used to specify timeouts if customers want it.
> 
> They can either pass in a number or a "TimeoutOptions" in this parameter to control connect, execute and close timeouts granularly.

oh ok thanks @serramatutu ! so based on [this](https://github.com/dbt-labs/semantic-layer-sdk-python/tree/main/.changes/unreleased), do we want to say:

> Added a new `timeout` parameter to Semantic Layer client and to the underlying GraphQL clients to specify timeouts. Set timeout number or `total_timeout` in the global `TimeoutOptions` to control connect, execute and close timeouts granularly. ExponentialBackoff.timeout_ms` is now deprecated.

mirnawong1 committed:
Merge branch 'current' into python-timeout

serramatutu commented:
yes!

mirnawong1 committed:
Update release-notes.md

update rn

mirnawong1 commented:
great stuff, thanks so much! merging @serramatutu !

mirnawong1 commented on website/docs/docs/dbt-versions/release-notes.md:
```diff
@@ -18,6 +18,7 @@ Release notes are grouped by month for both multi-tenant and virtual private clo
 
 ## February 2025
 
+- **Enhancement**:  The [Python SDK](/docs/dbt-cloud-apis/sl-python) added a new timeout parameter to Semantic Layer client and to the underlying GraphQL clients to specify timeouts. Set timeout number or `total_timeout` in the global `TimeoutOptions` to control connect, execute and close timeouts granularly. `ExponentialBackoff.timeout_ms` is now deprecated.
```
```suggestion
- **Enhancement**:  The [Python SDK](/docs/dbt-cloud-apis/sl-python) added a new timeout parameter to Semantic Layer client and to underlying GraphQL clients to specify timeouts. Set timeout number or `total_timeout` in the global `TimeoutOptions` to control connect, execute and close timeouts granularly. `ExponentialBackoff.timeout_ms` is now deprecated.
```

mirnawong1 committed:
Update website/docs/docs/dbt-versions/release-notes.md

mirnawong1 committed:
Update release-notes.md
